### PR TITLE
injecting funny reagent into a cigarret now requires a specific player status

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
injecting funny reagent into a cigarret now requires a specific player status
* be an antagonist
* OR has 3 hours living playtime

## Why It's Good For The Game
if they are new account, using such mechanics shouldn't be general when especially it's obvious for griefing

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.
image

image

----------

when you have antag datum or has playtime enough

image

image

</details>

## Changelog
:cl:
add: injecting some reagent into a cigarret now requires a specific player status. you need to be an antagonist or should have living playtime enough.
server: new config 'USE_EXP_TRACK_FOR_ANTAGONISM_BEHAVIOR' to check non-antag players have living time enough
/:cl:

### CODE REVIEW

**Comment by mystery3525** [comment]
Bandaid solution along with setting bad precedent for the ability to lock certain things

File: code/__HELPERS/antag_check.dm [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+		play_records = owner.prefs.exp
+		if (!play_records.len)
+			return TRUE // having nothing seems DB is broken. returns TRUE anyway.
+	var/playing_time = play_records[EXP_TYPE_LIVING]
+	if(playing_time >= req_time)
+		message_admins("[ADMIN_LOOKUPFLW(M)] tried '[behavior]' - they are not antag, but accepted with [playing_time] minutes playing time.")
+		log_admin("[key_name(M)] tried '[behavior]' - they are not antag, but accepted with [playing_time] minutes playing time.")
+		return TRUE
+
+	message_admins("[ADMIN_LOOKUPFLW(M)] tried '[behavior]', but it is rejected as they are not antag, and have not enough playtime. ([playing_time] minutes)")
```
Comment (Tsar-Salat) [inline code/__HELPERS/antag_check.dm]:
```suggestion
	message_admins("[ADMIN_LOOKUPFLW(M)] tried '[behavior]', but it is rejected as they are not antag, and are below the minimum playtime threshold. ([playing_time] minutes)")
```

File: code/__HELPERS/antag_check.dm [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	log_admin("[key_name(M)] tried '[behavior]', but it is rejected as they are not antag, and have not enough playtime. ([playing_time] minutes)")
```
Comment (Tsar-Salat) [inline code/__HELPERS/antag_check.dm]:
```suggestion
	log_admin("[key_name(M)] tried '[behavior]', but it is rejected as they are not antag, and are below the minimum playtime threshold. ([playing_time] minutes)")
```

**Comment by Tsar-Salat** [comment]
i dunno why that split into two reviews

**Comment by Fronsis** [comment]
In fact the hour requirement could be  bumped if admins wishes, a three hour requirement it's basically one round and a half to unlock it.

Wouldn't like to see iconic plasma cig go away since it's used by certain antag players, but for sure we need something to stop temporally these griefers, ban list speak for itself.

**Comment by EvilDragonfiend** [comment]
> Back before supercruise, if you were stuck on a space ruin with no jetpack. You would take a cigarette, put 5u of welding fuel it in, walk a tile forward, light it and be blasted across space into the next map. This was a legitimate strategy when you didn't know where the fuck you were and had some hit points to spare
> 
> That's just an example, but you get the picture.

I know what people are concerned about, but those things are not what new players. think and use. that's why the minimum time isn't that high.

**Comment by Tsar-Salat** [comment]
> that's why the minimum time isn't that high.

I was responding to fronsis. I said I agreed with the time you set

**Comment by Autisem** [comment]
I think it would be better to have a unique admin log instead so admins can be made aware that its probably griefer instead of locking it, they can then deal with it

**Comment by ro5490** [comment]
> I think it would be better to have a unique admin log instead so admins can be made aware that its probably griefer instead of locking it, they can then deal with it

Adding reagents is already logged so adding antag-only/playtime lock just sets a precident for locking game mechanics.
If we're going to lock things to playtime or similar its just going to end up with so many things locked out, hell, a new player can join as a chemist as it is currently and just create grenades

I dont think this is the best route to take, its already logged.
Maybe have a longer action timer for doing anything with a cig that has chemicals loaded in or similar might work, as it stands, cigs can be planted on someone and lit within a second or two, not giving any reaction time at all

**Comment by Tyranicranger4** [comment]
This isn't the solution here.

**Comment by EvilDragonfiend** [comment]
like Autisem said, this can be used to check specific situations rather than restricting.

**Comment by FriendlyContractor** [comment]
While time requirement for a very niche and specific feature is a horrible band aid, non-antagonists have no need to spike cigarettes with any chemicals and feature only encourages and enables players to grief. I stand by the opinion that mechanically locking features like to antagonists only will greatly benefit the server.

**Comment by EvilDragonfiend** [comment]
the actual issue is that beestation has low staff if we talk about the real solution.

**Comment by Fronsis** [comment]
I'm curious since a lot of admins have pointed and reacted on this PR that this is not the solution, we're all aware of the staff shortage Bee always had so i won't touch that, i ask what's inherently wrong with this temporally solution? Shouldn't this prevent griefers to easily access an insta-grief mechanic until the ''meta'' dies off?  though i might answer myself with this: I think i understand that if we start with this then people will argue to add hours for Chem nades, Scientists TTV manipulation, etc. 

Honestly i don't blame this PR, because we've been having the same couple of griefers almost 2 weeks now(And before anyone mentions it, i've done player reports, ahelps and discords pings in different ocassions) and sadly and understandable so Admins aren't online 24/7 to monitor any individual that dips a cig with WF/Plasma,  I think there's always the famous Bunker mode but i think that's used under very specific circumstances.

In any case i could also see the suggestion Autisme said a log-warn for the current online admin about the situation but i hope that's accompanied with a more will to aheal the affected individuals that manage to get caught in the grief before admin take cares of the matter because the WF/Plasma destroy most of your belongings and sometimes delimbs you. (Mostly commenting it because i've seen different griefs situations through the years were aheals were and weren't offered even when it was a big impact grief)

But all in all i mostly speak from a player perspective since i don't have admin/coding experience, just wanted to point out the POV of an average member regarding this PR, hence why i asked(and maybe even answered mysef lol)

**Comment by Tyranicranger4** [comment]
> Bee always had so i won't touch that, i ask what's inherently wrong with this temporally solution? Shouldn't this prevent griefers to easily access an insta-grief mechanic until the ''meta'' dies off?

This griefer "meta" is actually somewhat ideal for us all things considered. It's fairly blatant and much lower yield than most other tactics employed by griefers. If they can't put reagents in a cig to blow someone up then there is a very legitimate chance they'll opt to move on to more destructive tactics that they aren't blocked from performing. And blocking *all* of those other tactics for new accounts isn't really practical either.

**Comment by LodedDiper69** [comment]
If we really want to stop griefers we just add a whitelist. Been demanding this for months but everyone hates that idea I guess.

**Comment by Tsar-Salat** [comment]
> we really want to stop griefers we just add a whitelist.

If our primary concern is stopping griefers, then yes, that is a viable method. However, stopping griefers is not our primary concern.

The way whitelisted servers like skyrat work is that they offer a unique design, and have a sizeable playerbase that were both willing and commited enough to get whitelisted.

For bee though? The server thats struggling enough with pop? People would just go to MONKE, who already have supercruise and what makes us special

**Comment by EvilDragonfiend** [comment]
closing PR to improve the basic idea more

**Pull request closed without merging**